backend/tournesol/views/exports.py

```diff
@@ -60,40 +60,16 @@ def write_public_comparisons_file(request, write_target):
     ]
     writer = csv.DictWriter(write_target, fieldnames=fieldnames)
     writer.writeheader()
-    public_data = (
-        ContributorRating.objects
-        .filter(is_public=True, poll__name=DEFAULT_POLL_NAME)
-        .select_related("user", "entity")
-    )
-    serialized_comparisons = []
-    public_videos = set((rating.user, rating.entity) for rating in public_data)
-    comparisons = (
-        Comparison.objects
-        .filter(poll__name=DEFAULT_POLL_NAME)
-        .select_related("entity_1", "entity_2", "user")
-        .prefetch_related("criteria_scores")
-    )
-    public_comparisons = [
-        comparison
-        for comparison in comparisons
-        if (
-            (comparison.user, comparison.entity_1) in public_videos
-            and (comparison.user, comparison.entity_2) in public_videos
-        )
-    ]
-    public_usernames = [comparison.user.username for comparison in public_comparisons]
-    serialized_comparisons = ComparisonSerializer(public_comparisons, many=True).data
     writer.writerows(
         {
-            "public_username": public_username,
-            "video_a": comparison["entity_a"]["uid"].split(UID_DELIMITER)[1],
-            "video_b": comparison["entity_b"]["uid"].split(UID_DELIMITER)[1],
-            **criteria_score,
+            "public_username": comparison.username,
+            "video_a": comparison.uid_a.split(UID_DELIMITER)[1],
+            "video_b": comparison.uid_b.split(UID_DELIMITER)[1],
+            "criteria": comparison.criteria,
+            "weight": comparison.weight,
+            "score": comparison.score,
         }
-        for (public_username, comparison) in zip(
-            public_usernames, serialized_comparisons
-        )
-        for criteria_score in comparison["criteria_scores"]
+        for comparison in get_dataset(poll_name)
```

Comment:
An iterator can be used here to reduce the memory usage while writing the csv content:
```suggestion
        for comparison in get_dataset(poll_name).iterator()
```